services/api/src/messaging/players.service.ts

```diff
@@ -232,6 +299,42 @@ export const subscribe = (ws: GameWebSocket, newPlayer: Player): void => {
 
 export const unsubscribe = (ws: GameWebSocket): void => {
   const players = getPlayers(ws.roomId);
+  try {
+    const playerId = findPlayerByConnection(ws).id;
+
+    if (
+      gameService.getVoters(ws.roomId).length === 2 &&
+      gameRoomService.getGameState(ws.roomId) !== GameState.GAME_FINISHED
```

Comment:
Can we create a general algorithm that would be applied no matter how many players we have in the game or what the game state is.